### PR TITLE
wallet: Add missing check for -descriptors wallet tool option

### PR DESCRIPTION
**Pull request by maflcko**
Also, fix a test failure when compiled without sqlite

### CODE REVIEW

**Review by ryanofsky — APPROVED** [review]
Code review ACK fae32f295cc5b57c1cb95090bb60cddb42f9778a. Thanks for implementing the -descriptors check and dealing with the test failure!

File: test/functional/tool_wallet.py [hunk 1]
```diff
@@ -344,11 +344,13 @@ def test_dump_createfromdump(self):
         self.assert_raises_tool_error('Dump file {} does not exist.'.format(non_exist_dump), '-wallet=todump', '-dumpfile={}'.format(non_exist_dump), 'createfromdump')
         wallet_path = os.path.join(self.nodes[0].datadir, 'regtest/wallets/todump2')
         self.assert_raises_tool_error('Failed to create database path \'{}\'. Database already exists.'.format(wallet_path), '-wallet=todump2', '-dumpfile={}'.format(wallet_dump), 'createfromdump')
+        self.assert_raises_tool_error("The -descriptors option can only be used with the 'create' command.", '-descriptors', '-wallet=todump2', '-dumpfile={}'.format(wallet_dump), 'createfromdump')
```
Comment (jonatack) [inline test/functional/tool_wallet.py:347]: style nit, can use f-strings if you retouch

File: src/wallet/wallettool.cpp [hunk 2]
```diff
@@ -103,25 +103,27 @@ static void WalletShowInfo(CWallet* wallet_instance)
...
         return false;
     }
-    // -dumpfile is only allowed with dump and createfromdump. Disallow it for all other commands.
-    if (gArgs.IsArgSet("-dumpfile") && command != "dump" && command != "createfromdump") {
+    if (args.IsArgSet("-dumpfile") && command != "dump" && command != "createfromdump") {
         tfm::format(std::cerr, "The -dumpfile option can only be used with the \"dump\" and \"createfromdump\" commands.\n");
         return false;
     }
+    if (args.IsArgSet("-descriptors") && command != "create") {
+        tfm::format(std::cerr, "The -descriptors option can only be used with the 'create' command.\n");
```
Comment (jonatack) [inline src/wallet/wallettool.cpp:119]: the other error messages use double quotes around the command names (lines 112, 117)
```suggestion
        tfm::format(std::cerr, "The -descriptors option can only be used with the \"create\" command.\n");
```

**Review by jonatack — COMMENTED** [review]
Code review utACK fae32f295cc5b57c1cb95090bb60cddb42f9778a

File: src/wallet/wallettool.cpp (hunk 2 above)
Comment (MarcoFalke) [reply src/wallet/wallettool.cpp:119]: the help text uses single quotes

Comment (jonatack) [reply src/wallet/wallettool.cpp:119]: yup, i noticed the help follows a different format than the errors

**Pull request merged**